src/views/DatasetManager/components/DatasetOverview/components/CategoryAccordion/components/CategoryDetailsDialog/CategoryDetailsDialog.js

```diff
@@ -0,0 +1,157 @@
+                  </Typography>
+                  <Grid item sx={{ mt: 3 }}>
+                    <KPIValue
+                      kpi={kpis.find((kpiResult) => kpiResult.id === kpi.id)}
+                      valueTypographyProps={{ variant: 'body2', sx: { opacity: '70%' } }}
+                    />
+                  </Grid>
+                </CardContent>
+              </Card>
+            </Grid>
+          );
+        })}
+      </Grid>
+    );
+  }, [category.id, category.kpis, kpis, t]);
+
+  const query = useMemo(
+    () => getQuery(category?.previewTable?.queryId)?.query,
+    [category?.previewTable?.queryId, getQuery]
+  );
+
+  const tableEdit = useMemo(() => {
+    if (open === false || query == null) return null;
+    const context = {
+      isDarkTheme,
+      editMode: false,
+      targetDatasetId: datasetId,
+    };
+
+    const parameterData = {
+      id: category?.id,
+      options: {
+        columns: category?.previewTable?.columns,
+        dynamicValues: {
+          query,
+          resultKey: 'fields',
+        },
+      },
+    };
+
+    const parameterValue = { status: TABLE_DATA_STATUS.EMPTY };
+
+    return (
+      <GenericTable
+        parameterData={parameterData}
+        context={context}
+        parameterValue={parameterValue}
+        setParameterValue={() => {}}
+        resetParameterValue={() => {}}
+      />
+    );
+  }, [query, open, isDarkTheme, datasetId, category?.id, category?.previewTable?.columns]);
+
+  return query == null ? null : (
+    <>
+      <Button data-cy="category-details-open-button" variant="outlined" onClick={handleOpen}>
+        {t('commoncomponents.dialog.share.dialog.buttons.open', 'Open')}
+      </Button>
+      <Dialog
+        disableEnforceFocus
+        fullScreen
+        open={open}
+        onClose={handleClose}
+        fullWidth={true}
+        maxWidth={'xl'}
+        PaperProps={{
+          style: {
+            minHeight: '100%',
+            maxHeight: '100%',
+          },
+        }}
+      >
+        <DialogTitle>
+          {errorBanner}
+          <Button onClick={handleClose}>{t('commoncomponents.dialog.share.dialog.buttons.close', 'Close')}</Button>
+          <Box sx={{ flexDirection: 'column', alignItems: 'center', display: 'flex' }}>
+            <Box>
+              <Typography variant="h4">{datasetName}</Typography>
+              <Typography variant="h5" sx={{ opacity: '70%' }}>
+                {t(TranslationUtils.getDatasetCategoryNameTranslationKey(category.id), category.id ?? 'category')}
+              </Typography>
+            </Box>
+          </Box>
+        </DialogTitle>
+        <DialogContent>
+          {categories}
+          {tableEdit}
+        </DialogContent>
+      </Dialog>
+    </>
+  );
+};
+
+CategoryDetailsDialog.propTypes = {
+  category: PropTypes.object.isRequired,
+  kpis: PropTypes.array,
+};
+
+CategoryDetailsDialog.defaultProps = {};
```

Comment:
You're calling the `map` and `find` methods on the `kpi` prop without checking if it's set, maybe we should set an empty array as default value to prevent some errors.